### PR TITLE
Fix default response when --pipeline argument is used

### LINKED ISSUE #725: Empty Visualization when using --pipeline <pipeline_name>
## Description
When trying to visualize a specific pipeline, no visualization is created.

## Context
How has this bug affected you? What were you trying to accomplish?
I would like to be able to visualize the various sub-pipelines in my project

## Steps to Reproduce

1. create pipelines 
2. register pipelines
3. run pipelines
4. kedro viz --pipeline <pipeline_name>

## Expected Result
A visual of the named pipeline should appear in the browser window

## Actual Result
kedro viz tab opens with no visualized pipeline. If I do `kedro viz` then my default pipeline shows up fine. If I do `kedro viz --pipeline __default__`, it works. In fact, any pipeline can be visualized if I change it's key in the pipeline registry to '__default__'.
If I run `kedro viz --pipeline <pipeline_name>`, open blank visual in chrome ctrl-c in command window then run `kedro viz` it will show me the previously named pipeline rather than the default pipeline.

```
-- If you received an error, place it here.
```
INFO:     127.0.0.1:49846 - "GET /manifest.json HTTP/1.1" 404 Not Found
```
-- Separate them if you have more than one.
```

## Your Environment
Include as many relevant details as possible about the environment you experienced the bug in:

* Web browser system and version: Chrome Version 97.0.4692.99 (Official Build) (64-bit)
* Operating system and version: Windows Server 2016
* NodeJS version used (if relevant):
* Kedro version used (if relevant): 0.17.6
* Python version used (if relevant): 3.7.1
* Kedro Viz version: 4.2.0

## Checklist
- [ ] Include labels so that we can categorise your issue
- [ ] Bug

### PR DESCRIPTION
**Pull request by AntonyMilneQB**
## Description

Fixes #725: the `kedro viz --pipeline` CLI argument wasn't working correctly and would not load up any data for the pipeline graph.

With thanks to @tynandebold for his help investigating on the frontend side.

## Development notes

The `/pipelines/{registered_pipeline_id}` endpoint was working correctly, but we also need `/main/` to work correctly when there's only one registered pipeline selected. This response is provided by `get_default_response`, which was always looking for the `__default__` pipeline (`DEFAULT_REGISTERED_PIPELINE_ID`).

We now look for the correct pipeline with `data_access_manager.get_default_selected_pipeline().id` instead. This will default to `DEFAULT_REGISTERED_PIPELINE_ID` when present but use the selected pipeline otherwise (as happens when you have the `--pipeline` argument).

## QA notes

Tested on my own project and all works well now 🎉 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by tynandebold — APPROVED** [review]
Excellent!

**Review by limdauto — APPROVED** [review]
Nice one. Thank you!

**Comment by limdauto** [comment]
Nit: I feel like there might be an easy test we can write for this.

**Comment by AntonyMilneQB** [comment]
@limdauto I wanted to put in a test but couldn't see any tests for `get_default_response` already (which surprised me). Could you give a pointer to where these live?

**Comment by limdauto** [comment]
@AntonyMilneQB yes, that's where I'd put it. I'd create a sample fixture with a non-`__main__` default pipeline and check the response to not be empty.

**Review by limdauto — APPROVED** [review]
Amazing. Thank you

File: RELEASE.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Please follow the established format:
 - Migrate Kedro-UI buttons to Kedro-viz as Kedro-UI is now deprecated. (#716)
 - Add a Husky pre-push hook. (#723)
 - Create a `version` GraphQL query to get versions of Kedro-Viz. (#727)
+- Fix Kedro-Viz to work with projects that have no `__default__` registered pipeline. This also fixes the `--pipeline` CLI option to show the selected pipeline. (#729)
```
Comment (AntonyMilneQB) [inline RELEASE.md:14]:
```suggestion
- Fix Kedro-Viz to work with projects that have no `__default__` registered pipeline. This also fixes the `--pipeline` CLI option. (#729)
```

**Pull request merged**